src/Compilers/Server/VBCSCompiler/DesktopBuildServerController.cs

```diff
@@ -34,38 +28,35 @@ protected override IClientConnectionHost CreateClientConnectionHost(string pipeN
             // VBCSCompiler is installed in the same directory as csc.exe and vbc.exe which is also the 
             // location of the response files.
             var clientDirectory = AppDomain.CurrentDomain.BaseDirectory;
+#if NET46
             var sdkDirectory = RuntimeEnvironment.GetRuntimeDirectory();
             var compilerServerHost = new DesktopCompilerServerHost(clientDirectory, sdkDirectory);
+#else
+            var sdkDirectory = (string)null;
+            var compilerServerHost = new CoreClrCompilerServerHost(clientDirectory, sdkDirectory);
+#endif
             return new NamedPipeClientConnectionHost(compilerServerHost, pipeName);
         }
 
         protected internal override TimeSpan? GetKeepAliveTimeout()
         {
-            try
+            int keepAliveValue;
+            string keepAliveStr = _appSettings[KeepAliveSettingName];
+            if (int.TryParse(keepAliveStr, NumberStyles.Integer, CultureInfo.InvariantCulture, out keepAliveValue) &&
+                keepAliveValue >= 0)
             {
-                int keepAliveValue;
-                string keepAliveStr = _appSettings[KeepAliveSettingName];
-                if (int.TryParse(keepAliveStr, NumberStyles.Integer, CultureInfo.InvariantCulture, out keepAliveValue) &&
-                    keepAliveValue >= 0)
+                if (keepAliveValue == 0)
                 {
-                    if (keepAliveValue == 0)
-                    {
-                        // This is a one time server entry.
-                        return null;
-                    }
-                    else
-                    {
-                        return TimeSpan.FromSeconds(keepAliveValue);
-                    }
+                    // This is a one time server entry.
+                    return null;
                 }
                 else
                 {
-                    return ServerDispatcher.DefaultServerKeepAlive;
+                    return TimeSpan.FromSeconds(keepAliveValue);
                 }
             }
-            catch (ConfigurationErrorsException e)
```

Comment:
>catch (ConfigurationErrorsException e) [](start = 12, length = 38)

this needs to be kept. important case when the config file is corrupt

Reply:
can just catch exception if neccessary

---
In reply to: [138463943](https://github.com/dotnet/roslyn/pull/22075#discussion_r138463943) [](ancestors = 138463943)

Reply:
Wait. So the *indexer* on `NameValueCollection` throws on corruption, not when loading the config into `NameValueCollection` via `System.Configuration.ConfigurationManager.AppSettings`?